src/_guides/language/language-tour.md

```diff
@@ -3824,6 +3850,58 @@ and
 [Dart Language Asynchrony Support: Phase 2](/articles/language/beyond-async),
 and the [Dart language specification](/guides/language/spec).
 
+
+<a id="generator"></a>
+## Generators
+
+When you need to lazily produce a sequence of values,
+consider using a _generator function_.
+Dart has built-in support for two kinds of generator functions:
```

Comment:
Function should be singular in:
> for two kinds of generator function:

Reply:
Wow, this took me down a grammatical rathole! It turns out, opinions differ. The text and comments at [this grammar girl post](http://www.quickanddirtytips.com/education/grammar/sort-kind-or-type) imply that it should/can be plural (especially in US English). [This stack exchange topic](https://english.stackexchange.com/questions/29843/types-of-followed-by-singular-or-plural), on the other hand, is definitely on the "it should be singular" side. I'm going to leave it as is.